api/services/v1alpha1/monitoring_types.go

```diff
@@ -60,6 +60,52 @@ type Metrics struct {
 	Exporters map[string]string `json:"exporters,omitempty"`
 }
 
+// MetricsResources defines the resource requests and limits for the monitoring service
+type MetricsResources struct {
+	// CPULimit specifies the maximum CPU allocation (e.g., "500m", "2")
+	// +kubebuilder:default="1"
+	MonitoringStackCPULimit resource.Quantity `json:"monitoringStackCPULimit,omitempty"`
+	// CPURequest specifies the minimum CPU allocation (e.g., "100m", "0.5")
+	// +kubebuilder:default="100m"
+	MonitoringStackCPURequest resource.Quantity `json:"monitoringStackCPURequest,omitempty"`
+	// MemoryLimit specifies the maximum memory allocation (e.g., "1Gi", "512Mi")
+	// +kubebuilder:default="512Mi"
+	MonitoringStackMemoryLimit resource.Quantity `json:"monitoringStackMemoryLimit,omitempty"`
+	// MemoryRequest specifies the minimum memory allocation (e.g., "256Mi", "1Gi")
+	// +kubebuilder:default="256Mi"
+	MonitoringStackMemoryRequest resource.Quantity `json:"monitoringStackMemoryRequest,omitempty"`
+}
+
+type SharedResources struct {
+	// CollectorCPULimit specifies the maximum CPU allocation for the opentelemetry-collector (e.g., "1", "2")
+	// +kubebuilder:default="1"
+	CollectorCPULimit resource.Quantity `json:"collectorCPULimit,omitempty"`
+	// CollectorCPURequest specifies the minimum CPU allocation for the opentelemetry-collector (e.g., "100m", "0.5")
+	// +kubebuilder:default="100m"
+	CollectorCPURequest resource.Quantity `json:"collectorCPURequest,omitempty"`
+	// CollectorMemoryLimit specifies the maximum memory allocation for the opentelemetry-collector (e.g., "256Mi", "512Mi")
+	// +kubebuilder:default="256Mi"
+	CollectorMemoryLimit resource.Quantity `json:"collectorMemoryLimit,omitempty"`
+	// CollectorMemoryRequest specifies the minimum memory allocation for the opentelemetry-collector (e.g., "256Mi", "1Gi")
+	// +kubebuilder:default="256Mi"
+	CollectorMemoryRequest resource.Quantity `json:"collectorMemoryRequest,omitempty"`
```

Comment:
Here the request and limit is the same, I'm fine with that, but just want to be sure it is not a typo. Same for tempo.